database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -47,76 +41,95 @@ public void cleanup() {
 
     public int getCount() {
         return mSnapshots.size();
-
     }
+
     public DataSnapshot getItem(int index) {
         return mSnapshots.get(index);
     }
 
+    protected DataSnapshot internalGetItem(int index) {
```

Comment:
I found a bug here. When you remove the last element of the list, remove trigger events before and list have no element. So it would make an exception:
```
    java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
    at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
    at java.util.ArrayList.get(ArrayList.java:308)
    at com.firebase.ui.database.FirebaseArray.internalGetItem(FirebaseArray.java:61)
    at com.firebase.ui.database.FirebaseIndexArray.notifyChangedListeners(FirebaseIndexArray.java:119)
    at com.firebase.ui.database.FirebaseArray.notifyChangedListeners(FirebaseArray.java:119)
    at com.firebase.ui.database.FirebaseArray.onChildRemoved(FirebaseArray.java:100)
    at com.firebase.ui.database.FirebaseIndexArray.onChildRemoved(FirebaseIndexArray.java:29)
    at com.google.android.gms.internal.zzahe.zza(Unknown Source)
    at com.google.android.gms.internal.zzaje.zzcta(Unknown Source)
    at com.google.android.gms.internal.zzajh$1.run(Unknown Source)
    at android.os.Handler.handleCallback(Handler.java:739)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:155)
    at android.app.ActivityThread.main(ActivityThread.java:5727)
    at java.lang.reflect.Method.invoke(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1029)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
```

because `mSnapshots.remove(index)` was call before `notifyChangedListeners`
```
    @Override
    public void onChildRemoved(DataSnapshot snapshot) {
        int index = getIndexForKey(snapshot.getKey());
        mSnapshots.remove(index);
        notifyChangedListeners(OnChangedListener.EventType.Removed, index);
    }
```